src/agentscope/message/msg.py

```diff
@@ -1,168 +1,197 @@
 # -*- coding: utf-8 -*-
+# mypy: disable-error-code="misc"
 """The base class for message unit"""
-
-from typing import Any, Optional, Union, Literal, List
+from typing import (
+    Any,
+    Literal,
+    Union,
+    List,
+    Optional,
+)
 from uuid import uuid4
-import json
 
 from loguru import logger
 
-from ..utils.tools import _get_timestamp, _map_string_to_color_mark
+from ..serialize import is_serializable
+from ..utils.tools import (
+    _map_string_to_color_mark,
+    _get_timestamp,
+)
+
 
+class Msg:
```

Comment:
My concern is that making the `Msg`  no longer inheriting from `dict` in one PR may be too radical. Some existing implementations may be using `msg.get("content")` (like the [mention example](https://github.com/modelscope/agentscope/blob/21161fe9985ee2a2f617180b00a1424b81302d42/examples/conversation_with_mentions/main.py#L64)) and some developers may be used to this.

Reply:
Fix errors in the examples, and add a temporary solution, where Msg class supports `__getitem__` method with a warning.